cmd/ocm/create/upgradepolicy/cmd.go

```diff
@@ -81,85 +83,154 @@ func run(cmd *cobra.Command, argv []string) error {
+		Options: []string{"manual", "automatic"},
 	}
-	err = survey.AskOne(prompt, &version)
+	err = survey.AskOne(prompt, &scheduleType)
 	if err != nil {
-		return fmt.Errorf("Failed to get a valid version to upgrade to")
-	}
-	prompt = &survey.Select{
-		Message: "Schedule Upgrade",
-		Options: []string{"Upgrade now", "Schedule a different time"},
+		return fmt.Errorf("Failed to get a policy type")
 	}
-	err = survey.AskOne(prompt, &upgradePreference)
-	if err != nil {
-		return fmt.Errorf("Failed to get an upgrade time preference")
-	}
-	if upgradePreference == "Upgrade now" {
-		timestamp = time.Now().UTC().Add(time.Minute * 10)
 
-	} else {
+	var upgradeBuilder *cmv1.UpgradePolicyBuilder
 
-		var validationQs = []*survey.Question{
-			{
-				Name:   "date",
-				Prompt: &survey.Input{Message: "Please input desired date in format yyyy-mm-dd"},
-				Validate: func(val interface{}) error {
-					str, _ := val.(string)
-					_, err := time.Parse("2006-01-02", str)
-					if err != nil {
-						return fmt.Errorf("date format invalid")
-					}
-					return nil
-				},
-			},
-			{
-				Name:   "desiredTime",
-				Prompt: &survey.Input{Message: "Please input desired UTC time in format HH:mm"},
-				Validate: func(val interface{}) error {
-					str, _ := val.(string)
-					_, err := time.Parse("15:04", str)
-					if err != nil {
-						return fmt.Errorf("time format invalid")
-					}
-					return nil
-				},
-			},
+	if scheduleType == "automatic" {
+		prompt = &survey.Select{
+			Message: "Select day",
+			Options: []string{"Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"},
 		}
-		answers := struct {
-			Date        string
-			DesiredTime string
-		}{}
-		err = survey.Ask(validationQs, &answers)
+		var day string
+		err = survey.AskOne(prompt, &day)
 		if err != nil {
-			return err
+			return fmt.Errorf("Failed to get a valid day")
 		}
 
-		desiredTime := fmt.Sprintf("%sT%s:00.000Z", answers.Date, answers.DesiredTime)
-		timestamp, _ = time.Parse(time.RFC3339, desiredTime)
-		fmt.Println(timestamp)
-	}
+		var daysOfWeek = map[string]time.Weekday{
+			"Sunday":    time.Sunday,
+			"Monday":    time.Monday,
+			"Tuesday":   time.Tuesday,
+			"Wednesday": time.Wednesday,
+			"Thursday":  time.Thursday,
+			"Friday":    time.Friday,
+			"Saturday":  time.Saturday,
+		}
+
+		dayInt, ok := daysOfWeek[day]
+		if !ok {
+			return fmt.Errorf("Failed to get a valid day")
+		}
+
+		hours := make([]string, 24)
+		var start time.Time
+		for i := range hours {
+			t := start.Add(time.Hour * time.Duration(i))
+			hours[i] = t.Format("15:04")
+		}
+
+		prompt = &survey.Select{
+			Message: "Select hour",
+			Options: hours,
+		}
+		var hour string
+		err = survey.AskOne(prompt, &hour)
+		if err != nil {
+			return fmt.Errorf("Failed to get a valid hour")
+		}
+
+		hourInt, err := strconv.Atoi(strings.Split(hour, ":")[0])
+		if err != nil {
+			return nil
+		}
+
+		cronExpression := fmt.Sprintf("0 %d * * %d", hourInt, dayInt)
```

Comment:
The cron expression should be in UTC, while it seems that user might thinks the selection is on local time. This creates some deviation.